### DISCUSSION TITLE
Add delay for repetitive sending email in ASP .Net Identity

### DISCUSSION THREAD

**Discussion by mohabbati** [bitfoundation/bitplatform#1764] [General]
Hi all,

In **TodoTemplate** project, there are two methods that send email to the user for an email confirmation and reset password. To avoid repetitive sending emails, we need to add delay for it. I suggest the following implementation:

1. Add two properties for the last time email sent for "email confirmation" and "reset password" with the `DateTimeOffset` data type in the `User` class.
2. Extend the `UserManager<TUser>` class and then add two methods (one for email confirmation and another one for reset password) that return a `boolean` value that is it ok to send email?
3. The delay value store in the **appsetting.json**.
4. In the extended methods, calculate the time span between the last sent email time and `DateTimeOffset.Now` and compare it with the delay value. Finally, it returns a `bool` result.
5. The extended methods call in the `SendConfirmationEmail` and `SendResetPasswordEmail` methods. If the result is true then an email sends to the user otherwise not.

Next, in another refactoring we can move the `SendConfirmationEmail` and `SendResetPasswordEmail` methods to the `UserManagerExtensions` class

Note that by using the extension method instead of inheritance we have less refactoring.

What do you think about it?
@ysmoradi @msynk

**Comment by yasmoradi** [comment]
I prefer using separated methods or extension methods instead of inheriting from user manager @mohabbati

**Reply by mohabbati** [reply]
My point is extension methods as you said.
Could you explain what do you mean about "using separated methods"? @ysmoradi

**Reply by yasmoradi** [reply]
I mean write code directly in the controller's codes like the way we already did.
Someone who uses this template can move these codes to services or any place she would prefer.
